src/array.rs

```diff
@@ -136,43 +139,30 @@ impl<T: Miniconf, const N: usize> Miniconf for Array<T, N> {
     fn metadata() -> Metadata {
         let mut meta = T::metadata();
 
-        // Unconditionally account for separator since we add it
-        // even if elements that are deferred to (`Options`)
-        // may have no further hierarchy to add and remove the separator again.
-        meta.max_length += digits(N) + 1;
```

Comment:
This comment still seems relevant?

Reply:
The explanation is now much simpler. Added comment.